Rye/linux64 2

RyeMutt opened the pull request:
## Description

<!--
Describe what this PR changes or adds. Include any relevant context or motivation.
-->

## Related Issues

- [ ] **Please link to a relevant GitHub issue for additional context.**
  - **Bug Fix:** Link to an issue that includes reproduction steps and testing guidance.
  - **Feature/Enhancement:** Link to an issue with a write-up, rationale, and requirements.

Issue Link: <!-- e.g., closes #123 or relates to #456 -->

---

## Checklist

Please ensure the following before requesting review:

- [ ] I have provided a clear title and detailed description for this pull request.
- [ ] If useful, I have included media such as screenshots and video to show off my changes.
- [ ] The PR is linked to a relevant issue with sufficient context.
- [ ] I have tested the changes locally and verified they work as intended.
- [ ] All new and existing tests pass.
- [ ] Code follows the project's style guidelines.
- [ ] Documentation has been updated if needed.
- [ ] Any dependent changes have been merged and published in downstream modules
- [ ] I have reviewed the [contributing guidelines](https://github.com/secondlife/viewer/blob/develop/CONTRIBUTING.md).

---

## Additional Notes

<!--
Add any other information, screenshots, or suggestions for reviewers here.
-->

RyeMutt committed:
Fix usage of deprecate opengl defines

Signed-off-by: Rye <rye@alchemyviewer.org>

RyeMutt committed:
Fix deprecated opengl mipmap generation usage

Signed-off-by: Rye <rye@alchemyviewer.org>

RyeMutt committed:
Reduce extra packages installed during autobuild configure

Signed-off-by: Rye <rye@alchemyviewer.org>

RyeMutt committed:
Fix deprecated opengl draw commands

Signed-off-by: Rye <rye@alchemyviewer.org>

RyeMutt committed:
Rework anisotropic filtering options into a dropdown with levels for better performance tuning of low end gpu

Signed-off-by: Rye <rye@alchemyviewer.org>

RyeMutt committed:
Remove dead linux script

Signed-off-by: Rye <rye@alchemyviewer.org>

RyeMutt committed:
Fix cef scaling on linux hidpi displays

Signed-off-by: Rye <rye@alchemyviewer.org>

RyeMutt committed:
Fix broken mutex lock on skinned mesh map

Signed-off-by: Rye <rye@alchemyviewer.org>

RyeMutt committed:
Fix non-static LLCachedControl

Signed-off-by: Rye <rye@alchemyviewer.org>

RyeMutt committed:
Add early out for disk cache when app is shutting down to reduce shutdown time

Signed-off-by: Rye <rye@alchemyviewer.org>

RyeMutt committed:
Fix multiple unicode file io handling issues with llofstream and llifstream

Signed-off-by: Rye <rye@alchemyviewer.org>

RyeMutt committed:
Reduce texture cache start up time by reading header entries file in one call instead of many small ones

Signed-off-by: Rye <rye@alchemyviewer.org>

Geenz commented on indra/newview/llappviewer.cpp:
```diff
@@ -547,7 +547,7 @@ static void settings_to_globals()
 #endif
     LLRender::sNsightDebugSupport = gSavedSettings.getBOOL("RenderNsightDebugSupport");
-    LLImageGL::sGlobalUseAnisotropic    = gSavedSettings.getBOOL("RenderAnisotropic");
+    LLRender::sAnisotropicFilteringLevel = static_cast<F32>(gSavedSettings.getU32("RenderAnisotropicLevel"));
```
Does sAnisotropicFilteringLevel really need to be an F32?  I assume we need to cast somewhere down the line anyways - if so then I suppose this is fine.

Geenz commented on indra/llrender/llrender2dutils.cpp:
```diff
@@ -834,7 +834,7 @@ void gl_line_3d( const LLVector3& start, const LLVector3& end, const LLColor4& c
 
     gGL.flush();
-    glLineWidth(2.5f);
+    LLRender2D::setLineWidth(2.5f);
```
Why not the gGL equivalent?

RyeMutt replied:
Fixt

Geenz commented on indra/llrender/llrender.cpp:
```diff
@@ -524,9 +525,11 @@ void LLTexUnit::setTextureFilteringOptionFast(LLTexUnit::eTextureFilterOptions o
         {
-            glTexParameterf(sGLTextureType[tex_type], GL_TEXTURE_MAX_ANISOTROPY, gGLManager.mMaxAnisotropy);
+            F32 aniso_level = llclamp(LLRender::sAnisotropicFilteringLevel, 1.f, gGLManager.mMaxAnisotropy);
+            glTexParameterf(sGLTextureType[tex_type], GL_TEXTURE_MAX_ANISOTROPY, aniso_level);
```
Ah, I see what you're doing here now.  Disregard the previous casting comment.

RyeMutt committed:
Rework and modernize GL function/extension initialization for better wayland compatability

Signed-off-by: Rye <rye@alchemyviewer.org>

RyeMutt committed:
Remove dead linux crash logger

Signed-off-by: Rye <rye@alchemyviewer.org>

RyeMutt committed:
Fix invalid calls to glLineWidth on some linux GL drivers

Signed-off-by: Rye <rye@alchemyviewer.org>

RyeMutt committed:
Fix log to file not supporting multiline output

Signed-off-by: Rye <rye@alchemyviewer.org>

RyeMutt committed:
Fix threaded GL contexts on linux backend

Enable core profile context support on linux

Signed-off-by: Rye <rye@alchemyviewer.org>

RyeMutt committed:
Add support for GLX and EGL to GL extension loader
Fix dullahan wayland support
Fix shader GL_ERROR under GL Core profile in Mesa

Signed-off-by: Rye <rye@alchemyviewer.org>

Geenz committed:
Merge pull request #4827 from RyeMutt/rye/linux64

Linux Support Part 1
